uvicorn/config.py

```diff
@@ -280,11 +298,11 @@ def configure_logging(self):
             logging.getLogger("uvicorn.access").handlers = []
             logging.getLogger("uvicorn.access").propagate = False
 
-    def load(self):
+    def load(self) -> None:
         assert not self.loaded
 
-        if self.is_ssl:
-            self.ssl = create_ssl_context(
+        if self.is_ssl and self.ssl_certfile:
```

Comment:
why do we need to add this ` and self.ssl_certfile` this seems odd, is_ssl check if we have a keyfile or certfile already ?

Reply:
`certfile` is a required positional argument when running [`SSLContext.load_cert_chain`](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain).

`is_ssl` could return `True` if `certfile` is not present:

https://github.com/encode/uvicorn/blob/f025efd4a853e87921b07acdb6efb1d25931a686/uvicorn/config.py#L239-L242

Reply:
ok so the fix is wonrk imho since if someone forget to pass a certfile like `uvicorn dev.app:app --ssl-keyfile=dev/server.key` then the server will start but without ssl, which is deceiving
I'd propose instead :
```
        if self.is_ssl:
            assert self.ssl_certfile
            self.ssl = create_ssl_context(
```
so that the server does not start on that config mistake and it solves the mypy error
@Kludex ?

Reply:
I think the assertion would be reasonable here. I added it (8b38633), and I'm happy to consider alternatives also.